src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1671,6 +1671,51 @@ void CodeGen::genAvxFamilyIntrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_AVX512F_MoveMaskSpecial:
+        {
+            op1Reg            = op1->GetRegNum();
+            regNumber maskReg = node->ExtractTempReg(RBM_ALLMASK);
+
+            instruction maskIns;
+            instruction kmovIns;
+
+            switch (baseType)
```

Comment:
Might be worth a `TODO` covering that when we extend this to include other mask kinds, we'll need to update the handling.

That is, `TYP_BYTE` is `kmovq` for `VectorMask512<byte>`. But it is a `kmovd` for `VectorMask256<byte>` and a `kmovw` for `VectorMask128<byte>`